perf/src/cuda_runtime.rs

```diff
@@ -196,30 +209,49 @@ impl<T: Clone> PinnedVec<T> {
         self.x.as_mut_ptr()
     }
 
-    pub fn push(&mut self, x: T) {
+    fn prepare_realloc(&mut self, new_size: usize) -> (*mut T, usize) {
```

Comment:
@sakridge I'm late for the party after already merged... Yet, thanks for defining this helper function!